docs/dynamic_configuration.rst

```diff
@@ -1,46 +1,59 @@
-	they are applied asynchronously (upon the next wake up cycle) to every node, which gets subsequently reloaded.
-	If the node requires a restart to apply the configuration (for options with context postmaster or internal, if
-	their values have changed), a special flag indicating this should be set in the members.data JSON. A new API
-	endpoint should return whether the given node requires a restart. Additionally, the node status should also
-	indicate this.
+- Environment `configuration <https://github.com/zalando/patroni/blob/master/docs/ENVIRONMENT.rst>`
+	It is possible to set/override some of the "Local" configuration parameters with environment variables.
+	Environment configuration is very useful when you are running in dynamic environment and you don't know some of the parameters in advance (for example it's not possible to know you external IP address when you are running inside ``docker``).
+
+Some of PostgreSQL parameters must be set to the same value on master and replicas and therefore controlled only via "Dynamic configuration". Any attemt to change them via "Local configuration" are blocked:
 
-Some options may not be increased on the master independently of the replicas (master-dependent options):
+- max_connections: 100
+- max_locks_per_transaction: 64
+- max_worker_processes: 8
+- max_prepared_transactions: 0
+- wal_level: hot_standby
+- wal_log_hints: on
+- track_commit_timestamp: off
 
-- max_connections
-- max_locks_per_transactions
-- max_worker_processes
-- max_prepared_transactions
+Although following parameters are not necessarily must be the same on master and replicas, but it is better to have them eqals, therefore they are also controlled by Patroni and can be set only via Dynamic configuration.
+- max_wal_senders: 5
+- max_replication_slots: 5
+- wal_keep_segments: 8
 
-Regarding the options that can be set, the following restrictions apply:
+Important! Patroni does some simple checks of new values obtained from DCS before applying them. Above lists containing some sane default values and it is not allowed to set new value smaller.
 
-- dynamic configuration options that are also listed in the startup configuration will not be changed,
-  except for the case of master-dependent options.
+There are some other PostgreSQL parameters controlled by Patroni:
+- listen_addresses - is set either from ``postgresql.listen`` or from ``PATRONI_POSTGRESQL_LISTEN`` environment variable
+- port - is set either from ``postgresql.listen`` or from ``PATRONI_POSTGRESQL_LISTEN`` environment variable
+- cluster_name - is set either from ``scope`` or from ``PATRRONI_SCOPE`` environment variable
+- hot_standby: on
 
-When applying the startup or dynamic configuration options, the following actions should be taken:
+To be on the safe side parameters from the above lists are not written into ``postgresql.conf``, but passed as a list of arguments to the ``pg_ctl start`` what makes it not possible to change them even with 'ALTER SYSTEM'
```

Comment:
I'd change it to:

"The parameters from the list above are crucially important for the normal functioning of Patroni. Their values cannot be changed at the arbitrary point in time, or cannot be changed at all. Therefore, Patroni passes their values directly to the  ``pg_ctl start``, overriding all attempts to modify them via the PostgreSQL  'ALTER SYSTEM' command."